### ISSUE TITLE
Aliasing Checker : Returning @Unique object does not raise an error

### ISSUE THREAD

**Issue by aditya3434** [typetools/checker-framework#3486]
Consider the code given below.

```
@Unique
class Data {

    Object check(Data p) { return p; }   // No error

    Object check2(@Unique Data p) { return p; }   // No error

}
```

When the code is compiled by the Aliasing Checker, it doesn't raise any error even when the object is explicitly annotated. 

Shouldn't it ideally raise an error in the ```return p``` line, since ```Data p``` is ```@Unique``` and is now leaked?
